### PR TITLE
Make having app.json optional [again]

### PR DESCRIPTION
**Pull request by matthew-andrews**
Please consider :grinning:

### CODE REVIEW

File: commands/init.js [hunk 1]
```diff
@@ -52,7 +53,6 @@ function createDockerfile(dir) {
 function createDockerCompose(dir) {
   var composeFile = path.join(dir, docker.composeFilename);
   var procfile = directory.readProcfile(dir);
-  var appJSON = JSON.parse(fs.readFileSync(path.join(dir, 'app.json'), { encoding: 'utf8' }));
```
Comment (matthew-andrews) [inline commands/init.js]: this seemed to duplicate line 65

**Comment by matthew-andrews** [comment]
Or maybe read app json should be a property of the directory module ?

**Comment by hunterloftis** [comment]
One thing to consider here is that, without an app.json, `docker:init` won't work (it won't know what image to provide).

If we're going to do this, I'd like to do it without totally disabling one of the commands... one option to make app.json optional while leaving `:init` intact might be to require an 'image' from somewhere else, like `heroku docker:init --image heroku/nodejs`. If `--image` is not provided, we look to app.json for that data. If it can't be found in app.json, then we error out.

How does that sound?

**Comment by hunterloftis** [comment]
@matthew-andrews take a look at https://github.com/heroku/heroku-docker/pull/121 to see if that will be satisfactory for you folks

**Pull request closed without merging**